src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User register(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isEmpty()) {
+            User newUser = user.get();
+            newUser.setPassword(password);
+            newUser.setEmail(email);
+            return userService.add(newUser);
+        }
+        throw new AuthenticationException("User with such email already exist");
```

Comment:
Why you throw Auth exception in register method? Method signature must be `User register(String email, String password);`

Reply:
++